### DISCUSSION TITLE
Recontextualized getting started

### DISCUSSION THREAD

**Discussion by eisenhaus335** [fastify/fastify#3256] [Q&A]
Hi. I recently just started learning fastify and I found issue with how "getting started" and "encapsulation" explain the fundamental of fastify context and plugin. This is from my perspective as who is new in fastify. Maybe with this discussion, we could find clarity on how to understand this

So the getting started introduced the concept of "everything is a plugin". I take it as it is and understand that with this `register` does  register a plugin that do something. Everything from routes, db-connector and so on. But, no emphasis about routes and db-connector, it might need to be add that both is "plugin" that function as routes and db-connector. And by doing that, we are register them as plugin (of that root context)

The next problem is that cause contradiction of previous section in "encapsulation" tutorial. That section explain new concept of "context" and "plugin". But we doesn't know 'register' make other things. That is it make "context" and "plugin". It is contradictory of previous sentence. That cause confusion due the "encapsulation" make implied that by using `register` it make both "context" and "plugin. That was not explained or declared and that cause sort of vacuum, where you dont know what `register` does. Is it make plugin or "plugin and context"

So far i think this what i could found. Maybe someone can explain it. I understand them now, but maybe this could be re examined so there is clarity in the tutorial. So no one will stuck in understanding "context" and plugin. But, learning fastify is great and i have great experience learning from it

**Comment by jsumners** [comment]
> The next problem is that cause contradiction of previous section in "encapsulation" tutorial. That section explain new concept of "context" and "plugin". But we doesn't know 'register' make other things. That is it make "context" and "plugin". It is contradictory of previous sentence. That cause confusion due the "encapsulation" make implied that by using `register` it make both "context" and "plugin. That was not explained or declared and that cause sort of vacuum, where you dont know what `register` does. Is it make plugin or "plugin and context"

Please try re-wording this. I am unable to follow it and thus do not understand what the issue is.
